server/drivers/hd44780.c

```diff
@@ -877,6 +1128,67 @@ HD44780_set_brightness(Driver *drvthis, int state, int promille)
 	p->backlightstate = -1;
 }
 
+/** Sets internal backlight state using internal commands (based on configured model) */
+static void
+hd44780_set_backlight_internal(PrivateData *p, int state)
+{
+	unsigned char cmd = 0;
+	int brightness = state ? p->brightness : p->offbrightness;
+	switch (p->model) {
+		case HD44780_MODEL_WINSTAR_OLED:
+			cmd = WINST_MODESET | WINST_TEXTMODE \
+			    | (brightness >= MAX_BRIGHTNESS/2 ? WINST_PWRON : WINST_PWROFF);
+			p->hd44780_functions->senddata(p, 0, RS_INSTR, cmd);
+			break;
+
+		case HD44780_MODEL_PT6314_VFD:
+			cmd =  p->func_set_mode & ~PT6314_BRIGHT_MASK;
+			if (brightness >= 3 * MAX_BRIGHTNESS / 4)
+				cmd |= PT6314_BRIGHT_100; /* = 0x00 */
+			else if (brightness >= MAX_BRIGHTNESS / 2)
+				cmd |= PT6314_BRIGHT_75; /* = 0x01 */
+			else if (brightness > MAX_BRIGHTNESS / 4)
+				cmd |= PT6314_BRIGHT_50; /* = 0x02 */
+			else
+				cmd |= PT6314_BRIGHT_25; /* = 0x03 */
+			p->hd44780_functions->senddata(p, 0, RS_INSTR, cmd);
+			break;
+
+		default:
+			/* do nothing for other models */
+			break;
+	}
+}
+
+static void
+hd44780_set_backlight_config_cmds(PrivateData *p, int state)
+{
+	int brightness = state ? p->brightness : p->offbrightness;
+	int i, shift;
+	unsigned char cmd;
+
+	/* Assume two levels of brightness */
+	if (brightness >= MAX_BRIGHTNESS/2) {
+		for (i=0; i<sizeof(p->backlight_cmd_on); i++) {
+			shift = (sizeof(p->backlight_cmd_on) - i - 1)*8;
+			cmd =  (unsigned char)(p->backlight_cmd_on >> shift) & 0xff;
+			if (cmd) {
+				fprintf(stderr, "hd44780_set_backlight_config_cmds: sending BL on %02x\n", cmd);
```

Comment:
Please no calls to fprintf(stderr, ...) - either use the standard report()/debug() logging or don't write anything at all.

Reply:
Yes, it is just left-over after debugging.

But I would like to leave some other way of debugging this (when for example report level is debug or more), but I don't see a way without passing `Driver *` structure pointer to this function which I want to avoid.

Reply:
Sorry, just noticed change in report() API - I thought that this is called through drvthis->report() pointer. 

Now changed as you suggested.